src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+        this.ui = ui;
+        this.tasklist = tasklist;
+    }
+
+    /**
+     * Initialises the parser to start processing the parseMessage. Continues until user terminates.
+     * @return the response by Duke.
+     */
+    public String start() {
+
+        try {
+            String command = parseMessage.split(" ")[0];
+            String response = "";
+            if (command.equals("bye")) {
+                ByeCommand byeCommand = new ByeCommand();
+                response = byeCommand.execute(tasklist, ui);
+            } else if (command.equals("list")) {
+                ListCommand listCommand = new ListCommand();
+                response = listCommand.execute(tasklist, ui);
+            } else if (command.equals("find")) {
+                String wordToFind = this.parseMessage.substring(command.length());
+                FindCommand findCommand = new FindCommand(wordToFind);
+                response = findCommand.execute(tasklist, ui);
+            } else if (command.equals("done")) {
+                int index = Integer.parseInt(parseMessage.split(" ")[1]) - 1;
+                DoneCommand doneCommand = new DoneCommand(index);
+                response = doneCommand.execute(tasklist, ui);
+            } else if (command.equals("delete")) {
+                int index = Integer.parseInt(parseMessage.split(" ")[1]) - 1;
+                DeleteCommand deleteCommand = new DeleteCommand(index);
+                response = deleteCommand.execute(tasklist, ui);
+            } else if (command.equals("event") || command.equals("deadline") || command.equals("todo")) {
+                String dissectedMessage = this.parseMessage.substring(command.length());
+                response = commandTasks(command, dissectedMessage);
+            } else {
+                String errorMessage = "Command is wrong. Please start with delete, "
+                        + "list, done, deadline, todo or event.";
+                throw new DukeCommandException(errorMessage);
+            }
+            this.tasklist.updateStorage();
+            return response;
+        } catch (DukeCommandException | DukeIndexException | DukeTaskException | DukeListException e) {
+            return ui.printError(e.getMessage());
+        } catch (DateTimeParseException e) {
+            return ui.printError("INPUT DATE TIME FORMAT IS WRONG");
+        } catch (IOException e) {
+            return ui.printError(e.getMessage());
+        }
+    }
+
+    /**
+     * Helper method to organize the different types of task that can be recorded.
+     * @param tasktype Type of task to be recorded.
+     * @param message Details of the task.
+     * @return response from Duke.
+     * @throws DukeTaskException When there is an error in the input.
+     */
+    public String commandTasks(String tasktype, String message) throws DukeTaskException {
+
+        String[] parsedMessage = null;
+        Task newTask = null;
+        int size = this.tasklist.getTaskSize();
```

Comment:
Should this be more descriptive? maybe sizeTaskList

Reply:
Thanks!